py/torch_tensorrt/dynamo/conversion/impl/quantize.py

```diff
@@ -28,42 +29,59 @@ def quantize(
     """
 
     with unset_fake_temporarily():
-        if isinstance(input_tensor, TRTTensor) and input_tensor.dtype not in (
-            trt.float32,
-            trt.float16,
-        ):
-            raise ValueError(
-                f"quantize converter received an input of {input_tensor.dtype} type. Supported types: float32 | float16"
-            )
+        if isinstance(input_tensor, (torch.Tensor, TRTTensor)):
+            input_tensor = get_trt_tensor(ctx, input_tensor, name)
+            if input_tensor.dtype not in (
+                trt.float32,
+                trt.float16,
+            ):
+                raise ValueError(
+                    f"quantize converter received an input of {input_tensor.dtype} type. Supported types: float32 | float16"
+                )
             if num_bits != 8 or exponent_bits not in (0, 4):
                 raise ValueError(
                     f"quantize converter currently only accept INT8 or FP8 based quantize, got {num_bits=}, {exponent_bits=}"
                 )
+        else:
+            raise ValueError(
+                f"quantize converter received an input of {type(input_tensor)} type. Supported types: torch.Tensor | TRTTensor"
+            )
+
         if num_bits == 8 and exponent_bits == 0:
             max_bound = 127
         elif num_bits == 8 and exponent_bits == 4:
             max_bound = 448
 
-        amax = to_torch(amax, None)
-        scale = torch.divide(amax, max_bound)
-        scale = get_trt_tensor(ctx, scale, name + "_scale")
-        # Add Q node
-        quantize_layer = ctx.net.add_quantize(input_tensor, scale)
+        if not isinstance(amax, trt.ITensor):
+            # for the int8/fp8 quantization, it is per tensor quantization, so amax should be a singlular element
+            # TODO: to confirm with ModelOpt team, why I am getting a amax with shape > 1
+            if len(amax.shape) >= 1:
+                amax = amax.abs().amax()
+                if amax == 0.0:
+                    amax = 1.0
+            amax = to_torch(amax, None)
+            scale = torch.divide(amax, max_bound)
+            scale = get_trt_tensor(ctx, amax, name + "_scale")
```

Comment:
```suggestion
            amax = to_torch(amax, None)
            if len(amax.shape) >= 1:
                amax = amax.abs().amax()
                if amax == 0.0:
                    amax = 1.0
            scale = torch.divide(amax, max_bound)
            scale = get_trt_tensor(ctx, scale, name + "_scale")
```

`amax` needs to be a `torch.Tensor` before calling abs()
think you meant to do `get_trt_tensor(ctx, scale, name + "_scale")`?

Reply:
confirmed with modelopt interface, the input can only be a torch.Tensor, I have changed the interface to be torch.Tensor only removed np.ndarray